sklearn/neighbors/_nearest_centroid.py

```diff
@@ -132,8 +175,6 @@ def fit(self, X, y):
         else:
             X, y = self._validate_data(X, y, accept_sparse=["csr", "csc"])
         is_X_sparse = sp.issparse(X)
-        if is_X_sparse and self.shrink_threshold:
-            raise ValueError("threshold shrinking not supported for sparse input")
```

Comment:
So we should an entry in the changelog also to mention that we support the sparse data with the shrinking as well.